internal/cloudinfo/providers/amazon/config.go

```diff
@@ -40,29 +44,11 @@ type PricingConfig struct {
 }
 
 func (c Config) GetPricingCredentials() Credentials {
-	creds := c.Pricing.Credentials
-
-	if creds.AccessKey == "" {
-		creds.AccessKey = c.AccessKey
+	if reflect.DeepEqual(c.Pricing.Credentials, Credentials{}) {
```

Comment:
Technically, this is a breaking change compared to the previous fallback mechanism. While I don't really see why someone would want configure a different profile for example, but not different access credentials, it's still more correct than this comparison, so I'd like to see this change reverted.

Reply:
And how should I handle the AssumeRoleARN parameter? If we inherit a non-empty value in the pricing config, that means I can't achieve the following scenario:
 - in the base config I use AssumeRoleARN
 - in the pricing config I don't want to assume any role (and I might use different credentials)

Do you want me to inherit a non-empty AssumeRoleARN value or leave it out?

Probably it would help to understand why is there a need for a separate pricing creds config.
 
I'm okay to revert this, but I think the current mechanism can lead to unexpected behaviour, while my change creates clear outcomes.

Reply:
Fixed in 1825f649420210a99f8594fe5919ba4896d01322. I handle AssumeRoleARN as the other parameters.

Reply:
> Do you want me to inherit a non-empty AssumeRoleARN value or leave it out?

Fair point. Let's inherit it for now. The current configuration format is not ideal, so there is room for improvement here.

> Probably it would help to understand why is there a need for a separate pricing creds config.

Unfortunately the pricing API is not available in every partition, so if you want to run Cloudinfo in other partitions (GovCloud, China), you'll still need a set of credentials from the public partition which is why you can supply different credentials for pricing.